### PR TITLE
[BSE-4949] Support column selection in DataFrameGroupby getattr

### PR DESCRIPTION
**Pull request by chrisoh7**
## Changes included in this PR

As title.
```
df = pd.DataFrame({'Animal': ['Falcon', 'Falcon',
                              'Parrot', 'Parrot'],
                   'S': [380., 370., 24., 26.]})

print(df.groupby(['Animal']).S.mean())
```
Adds an additional try-except in getattr that runs getitem with `name` as key. On failure, raise original fallback warning and falls back. 

## Testing strategy

N/A

## User facing changes

Users can access groupby columns using simple attribute syntax. 

## Checklist
- [ ] Pipelines passed before requesting review. To run CI you must include `[run CI]` in your commit message.
- [x] I am familiar with the [Contributing Guide](https://github.com/bodo-ai/Bodo/blob/main/CONTRIBUTING.md) 
- [x] I have installed + ran pre-commit hooks.

### CODE REVIEW

File: bodo/pandas/groupby.py [hunk 1]
```diff
@@ -72,6 +72,11 @@ def __getattribute__(self, name: str, /) -> Any:
         try:
             return object.__getattribute__(self, name)
         except AttributeError:
+            pass
+        try:
```
Comment (scott-routledge2) [inline bodo/pandas/groupby.py]: We need to also validate that `attr` is a valid column in the DataFrame, otherwise, if someone does `grouped.apply`, we don't support apply yet, so it will think that "apply" is a column in the DataFrame as opposed to a method of DataFrameGroupBy. (And an edge case would be what happens if the DataFrame actually does have a column named "apply")

Comment (chrisoh7) [reply bodo/pandas/groupby.py]: Done.

File: bodo/tests/test_df_lib/test_end_to_end.py [hunk 2]
```diff
@@ -1970,6 +1972,39 @@ def test_series_describe():
...
+
+    df = pds.DataFrame({"apply": [1], "B": [1], "C": [2]})
+    bdf = bd.from_pandas(df)
+
+    grouped = bdf.groupby("B")
+
+    # Accessing a column: should not raise a warning
+    with warnings.catch_warnings(record=True) as record:
+        warnings.simplefilter("always")
+    assert not record, f"Unexpected warning when accessing column: {record}"
```
Comment (ehsantn) [inline bodo/tests/test_df_lib/test_end_to_end.py]: This doesn't run any code. Also, you need to use `pytest.warns`.

File: bodo/tests/test_df_lib/test_end_to_end.py [hunk 3]
```diff
@@ -1970,6 +1972,39 @@ def test_series_describe():
...
+
+    # Accessing an implemented Pandas GroupBy method: should raise fallback warning
+    with warnings.catch_warnings(record=True) as record:
+        warnings.simplefilter("always")
+    fallback_warnings = [
+        w for w in record if issubclass(w.category, BodoLibFallbackWarning)
+    ]
+    assert len(fallback_warnings) == 1
+    assert "apply" in str(fallback_warnings[0].message)
```
Comment (ehsantn) [inline bodo/tests/test_df_lib/test_end_to_end.py]: Same.

File: bodo/tests/test_df_lib/test_end_to_end.py (hunk 2 above)
Comment (chrisoh7) [reply bodo/tests/test_df_lib/test_end_to_end.py]: Oops, must have gotten deleted by accident.
Also I think pytest.warns can't be used when the intended behavior is not raising a warning. I did make the changes to use `pytest.warns` for the cases after this one.

**Review by scott-routledge2 — APPROVED** [review]
LGTM, thanks @chrisoh7 !
